### ISSUE TITLE
[a11y]: Breadcrumb focus restarts when tabbing from open overflow menu

### ISSUE THREAD

**Issue by warrenmblood** [carbon-design-system/carbon#19254]
### Package

@carbon/web-components

### Browser

Chrome

### Operating System

MacOS

### Package version

1.81.0

### Automated testing tool and ruleset

Manual

### Description

When navigating Breadcrumb with Overflow Menu using keyboard, after pressing tab from the open menu, the focus will return to the first breadcrumb link instead of continuing on to the next one. This adds extra steps for keyboard navigation and is inconsistent with the React version, which will continue through the links in order after opening a menu.

### WCAG 2.1 Violation

WCAG 2.1.1 (keyboard)

### Reproduction/example

https://web-components.carbondesignsystem.com/?path=/story/components-breadcrumb--breadcrumb-with-overflow-menu

### Steps to reproduce

1. Go to Breadcrumb With Overflow Menu storybook: https://web-components.carbondesignsystem.com/?path=/story/components-breadcrumb--breadcrumb-with-overflow-menu
2. Navigate (by pressing tab) to the ellipses (...) breadcrumb link with an overflow menu
3. Open the menu (space) and press tab. Focus will return to the first link instead of "Breadcrumb 5"

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
